opendrift/readers/reader_grib2.py

```diff
@@ -0,0 +1,158 @@
+#
+# OpenDrift is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with OpenDrift.  If not, see <https://www.gnu.org/licenses/>.
+#
+# Copyright 2020, Gaute Hope, MET Norway
+
+import logging
+logger = logging.getLogger(__name__)
+
+import numpy as np
+import xarray as xr
+from datetime import datetime
+
+from .basereader import BaseReader
+from .basereader.structured import StructuredReader
+
+class Reader(BaseReader, StructuredReader):
+    dataset = None
+    variables = None
+
+    def __init__(self, filename, proj4 = None, engine = 'cfgrib'):
+        """
+        Grib file reader
+
+        Args:
+
+            filename: path to grib or grib2 file.
+
+            proj4: Optional projection override.
+
+            engine: grib engine used by xarray, default is `cfgrib`.
+
+        Returns: Grib-file reader.
+
+        """
+        self.dataset = xr.open_dataset(filename, engine = engine)
+        self.name = filename
+
+        self.lat = self.dataset.latitude.values
+        self.lon = self.dataset.longitude.values
+        self.delta_x = np.abs(self.lon[0,1] - self.lon[0,0]).item()
+        print(self.delta_x)
+
+        self.xmin = self.dataset.longitude.min().item()
+        self.xmax = self.dataset.longitude.max().item()
+        self.ymin = self.dataset.latitude.min().item()
+        self.ymax = self.dataset.latitude.max().item()
+
+        self.times = self.dataset.time.values
+
+        # convert to python datetime
+        assert '[ns]' in self.times.dtype.str, "datetime is not in nanoseconds"
+        self.times = np.array([datetime.utcfromtimestamp(d.astype('int') / 1e9) for d in self.times])
+
+        self.start_time = self.times[0]
+        self.end_time = self.times[-1]
+        self.time_step = self.times[1] - self.times[0]
+        assert np.all(np.diff(self.times) == self.time_step), "time step not constant"
+
+        # scan for projection
+        self.proj4 = proj4
+        if self.proj4 is None:
+            for v in self.dataset.variables:
+                v = self.dataset[v]
+                if 'GRIB_gridType' in v.attrs:
+                    if self.proj4 is None:
+                        self.proj4 = v.attrs['GRIB_gridType']
+                        logger.info("projection: %s" % self.proj4)
+                    else:
+                        assert self.proj4 == v.attrs['GRIB_gridType'], "all variables must have the same projection"
+        else:
+            logger.info("Using supplied projection: %s" % self.proj4)
+
+        # parsing variables
+        self.variable_mapping = {}
+
+        for v in self.dataset.variables:
+            var = self.dataset[v]
+            std_name = var.attrs.get('GRIB_cfName', None)
```

Comment:
Is there a specific reason why 'std_name = var.attrs.get('GRIB_cfName', None)' instead of std_name = var.standard_name?

Reply:
No, I didn't realize that was available 🙂